src/mutation-observers/selector_observer.ts

```diff
@@ -7,17 +8,25 @@ export interface SelectorObserverDelegate {
   selectorMatchElement?(element: Element, details: object): boolean
 }
 
-export class SelectorObserver implements ElementObserverDelegate {
-  private selector: string
-  private elementObserver: ElementObserver
-  private delegate: SelectorObserverDelegate
-  private matchesByElement: Multimap<string, Element>
-  private details: object
-
-  constructor(element: Element, selector: string, delegate: SelectorObserverDelegate, details: object = {}) {
-    this.selector = selector
+export class SelectorObserver implements AttributeObserverDelegate, ElementObserverDelegate {
+  private readonly attributeObserver: AttributeObserver
+  private readonly elementObserver: ElementObserver
+  private readonly delegate: SelectorObserverDelegate
+  private readonly matchesByElement: Multimap<string, Element>
+  private readonly details: object
+  private selector: string | null
+
+  constructor(
+    element: Element,
+    attributeName: string,
+    scope: Element,
```

Comment:
This might be a bit nit-picky, but I feel like `scope` could be confused with the Stimulus [`Scope`](https://github.com/hotwired/stimulus/blob/47951cd011176f2c1a3dca0932d0c2bb79000892/src/core/scope.ts#L10) class here. 

Would it make sense to rename `element` to `controllerElement` and `scope` to `element`/`scopeElement`?